File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/repository/svn/SvnScmProviderRepository.java
package com.tencent.devops.scm.api.pojo.repository.svn;

import com.tencent.devops.scm.api.pojo.auth.IScmAuth;
import com.tencent.devops.scm.api.pojo.repository.ScmProviderRepository;
import lombok.Getter;

/**
 * svn提供者代码库信息
 */
@Getter
public class SvnScmProviderRepository implements ScmProviderRepository {

    public static final String CLASS_TYPE = "SVN";

    private String url;
    private String userName;
    private IScmAuth auth;

    public SvnScmProviderRepository withUrl(String url) {
        this.url = url;
        return this;
    }

    public SvnScmProviderRepository withUserName(String userName) {
        this.userName = userName;
        return this;
    }

    public SvnScmProviderRepository withAuth(IScmAuth auth) {
        this.auth = auth;
        return this;
    }
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/repository/svn/SvnScmProviderRepository.kt
package com.tencent.devops.scm.api.pojo.repository.svn

import com.tencent.devops.scm.api.pojo.auth.IScmAuth
import com.tencent.devops.scm.api.pojo.repository.ScmProviderRepository
import com.tencent.devops.scm.api.util.SvnUtils

data class SvnScmProviderRepository(
    val url: String,
    val userName: String? = null,
    val auth: IScmAuth? = null
) : ScmProviderRepository {
    val projectIdOrPath: Any = SvnUtils.getSvnProjectName(url)

    companion object {
        const val CLASS_TYPE = "SVN"
    }
}
